cloudplatform/connectivity-apache-httpclient5/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/ApacheHttpClient5Wrapper.java

```diff
@@ -70,7 +75,22 @@ public void close( final CloseMode closeMode )
         return httpClient.execute(target, wrapRequest(request), context);
     }
 
-    private ClassicHttpRequest wrapRequest( final ClassicHttpRequest request )
+    ApacheHttpClient5Wrapper withDestination( final HttpDestinationProperties destination )
+    {
+        // explicitly check the reference equality, since equals doesn't check header providers
+        // this is a slight improvement, avoiding unnecessary wrapper instantiation
+        // in cases where destination objects are reused / served from cache
+        if( !destination.equals(this.destination) ) {
```

Comment:
I'm confused, wouldn't "reference equality" mean...?

 ```suggestion
         // explicitly check the reference equality, since equals doesn't check header providers
         // this is a slight improvement, avoiding unnecessary wrapper instantiation
         // in cases where destination objects are reused / served from cache
         if( destination != this.destination ) {
```

Reply:
Nevermind, the comment is just misplaced.